paig-server/backend/paig/api/shield/scanners/AWSBedrockGuardrailScanner.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import boto3
+import os
+
+from api.shield.enum.ShieldEnums import Guardrail, RequestType
+from api.shield.model.scanner_result import ScannerResult
+from api.shield.model.analyzer_result import AnalyzerResult
+from api.shield.scanners.BaseScanner import Scanner
+
+logger = logging.getLogger(__name__)
+
+
+class AWSBedrockGuardrailScanner(Scanner):
+    """
+    Scanner implementation for applying guard rail policies in the input prompt.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initialize the GuardrailScanner with the specified parameters.
+
+        Parameters:
+            **kwargs: keyword arguments passed from properties file.
+            E.g.
+            name (str): The name of the scanner.
+            request_types (list): List of request types that the scanner will handle.
+            enforce_access_control (bool): Flag to enforce access control.
+            model_path (str): Path to the model used by the scanner.
+            model_score_threshold (float): Threshold score for the model to consider a match.
+            entity_type (str): Type of entity the scanner is looking for.
+            enable (bool): Flag to enable or disable the scanner.
+        """
+        super().__init__(**kwargs)
+
+        self.guardrail_id, self.guardrail_version, self.region = self.get_guardrail_details()
+        self.bedrock_client = boto3.client(
+            'bedrock-runtime',
+            region_name=self.region
+        )
+
+    def scan(self, message: str) -> ScannerResult:
+        """
+        Scan the input prompt through the Bedrock guardrail.
+
+        Parameters:
+            message (str): The input prompt that needs to be scanned.
+
+        Returns:
+            dict: Scan result including traits, actions, and output text if intervention occurs.
+        """
+        guardrail_source = Guardrail.INPUT.value if self.get_property('scan_for_req_type') in [
+            RequestType.PROMPT.value,
+            RequestType.ENRICHED_PROMPT.value,
+            RequestType.RAG.value
+        ] else Guardrail.OUTPUT.value
+        logger.debug(f"AWSBedrockGuardrailScanner: Scanning message: {message} with guardrail source: {guardrail_source} for {self.get_property('scan_for_req_type')}")
+
+        response = self.bedrock_client.apply_guardrail(
+            guardrailIdentifier=self.guardrail_id,
+            guardrailVersion=self.guardrail_version,
+            source=guardrail_source,
+            content=[{'text': {'text': message}}]
+        )
+        logger.debug(f"AWSBedrockGuardrailScanner: Response received: {response}")
```

Comment:
Can we set this to info level - the response does not contain any sensitive information, and this is helpful for debugging as the guard-rail response is not consistent